### PR TITLE
Add GET endpoint to read activation jobs

### PR DESCRIPTION
**Pull request by Dostonbek1**
Add a new API endpoint to retrieve jobs based on activation id.

Resolves: AAP-6201

### CODE REVIEW

File: src/eda_server/db/models/job.py [hunk 1]
```diff
@@ -35,6 +36,26 @@
     sa.Column("uuid", postgresql.UUID),
 )
 
+
+activation_job_instances = sa.Table(
```
Comment (cutwater) [inline src/eda_server/db/models/job.py:40]: Is it simply a many-to-many junction table?

File: src/eda_server/api/activation.py [hunk 2]
```diff
@@ -216,6 +216,21 @@ async def delete_activation(
     await db.commit()
 
 
+@router.get(
+    "/api/activations/{activation_id}/job_instances",
+    response_model=List[schema.ActivationJobInstance],
+    operation_id="list_activation_job_instances",
+)
+async def list_activation_job_instances(
```
Comment (cutwater) [inline src/eda_server/api/activation.py:224]: What's the point of returning only list of job instance IDs?

File: src/eda_server/db/models/job.py (hunk 1 above)
Comment (cutwater) [inline src/eda_server/db/models/job.py:40]: Since it's a new table, where it's populated?

**Review by cutwater — CHANGES_REQUESTED** [review]
Please add tests.

File: src/eda_server/db/models/job.py [hunk 3]
```diff
@@ -35,6 +36,26 @@
...
+    metadata,
+    sa.Column(
+        "id",
+        sa.Integer,
+        sa.Identity(always=True),
+        primary_key=True,
+    ),
+    sa.Column(
+        "activation_id",
+        sa.ForeignKey("activation.id", ondelete="CASCADE"),
```
Comment (cutwater) [inline src/eda_server/db/models/job.py:51]: Shouldn't it be `nullable=False`?

File: src/eda_server/db/models/job.py (hunk 1 above)
Comment (Dostonbek1) [reply src/eda_server/db/models/job.py:40]: I think it is one-to-many, activation can have multiple job instances but each job instance only belongs to one activation.

Comment (cutwater) [reply src/eda_server/db/models/job.py:40]: So why the extra table is needed then?

File: src/eda_server/api/activation.py (hunk 2 above)
Comment (Dostonbek1) [reply src/eda_server/api/activation.py:224]: I tried to follow the format that we already have for [Activation Instance Job Instances.](https://github.com/ansible/eda-server/blob/cbd41655e3ffe7da886a256ad8bfc2b1e3ad6504/src/eda_server/api/activation.py#L415) Is it wrong?

File: src/eda_server/db/models/job.py (hunk 1 above)
Comment (Dostonbek1) [reply src/eda_server/db/models/job.py:40]: For Activation Instance, its job instance table is populated on activation instance creation but for Activation, we currently don't have that function to populate its job instance table.

**Comment by Dostonbek1** [comment]
This PR was replaced with https://github.com/ansible/eda-server/pull/237.

**Comment by Dostonbek1** [comment]
This PR was replaced with https://github.com/ansible/eda-server/pull/237.

**Pull request closed without merging**